Install php as part of the setup.sh script

TheAspens opened the pull request:
Fixes issue #3

TheAspens commented:
@drshawnkwang can you test and merge if it works as intended?  Thanks!

Rytiss commented:
@TheAspens Why are you checking for existence of `apt-get` and then using `apt` instead?

TheAspens commented:
Because I wasn't being careful when I changed it from apt-get to apt.  Thanks for the catch.  Fixed.

drshawnkwang commented:
I did get this error, repeated for each package `composer` installed.

```
    Failed to download sebastian/version from dist: The zip extension and unzip command are both missing, skipping.
Your command-line PHP is using multiple ini files. Run `php --ini` to show them.
    Now trying to download from source
  - Installing sebastian/version (2.0.1): Cloning 99732be0dd from cache
```
The packages are correctly installed, but I think using git. On a bare (new) system, install package 'zip' and 'unzip'. Once these are installed, then composer won't spit out this warning. For Debian `apt install zip` will do the trick (as it installs unzip). For Redhat-family, I am unsure, but yum install unzip will probably do the trick.

drshawnkwang commented:
I finally got around to testing the updated `setup.sh` script. No problems, the PR is ready to be merged. However, two comments.

1. I don't have the write permissions to merge this PR. (FTR: I don't really think I need/want it.) You'll have to merge it.
1. Comment about `composer`. If you naively run the PHP commands on composer's Web site, you will get a executable named `composer.phar` in your current working directory. And `command -v composer` checks for the command 'composer' in ones $PATH. I think it would be worthwhile to add a comment to the script saying:

```
echo 'Please install composer with command "php composer-setup.php --install-dir=/usr/local/bin --filename=composer". Or manually move the executable composer.phar into your PATH and rename it "composer".'
```

Since I've used composer before I already knew I had to move the executable. Adding a sentence may make things more clear for people who have never downloaded composer before.

TheAspens commented:
@drshawnkwang - I just noticed this pull request lagging.  I believe I have addressed your concerns as well as fixed a few other issues I found (I have a new laptop so I'm fixing a few things).  Can you take a look and merge if possible?  You should have access to merge now.

drshawnkwang commented:
I tested this branch in a vagrant VM running vanilla Debian 9. Two comments

1. Github sats I do not have write access
1. When I run setup, everything works. When I try to start the boincserver with `ansible-playbook -i hosts start.yml` the command fails. I believe the problem is: boincserver starts an docker apache process which binds to port 80. However, as part of `setup.sh`, apache2 is installed in the VM. This starts the apache2 daemon and binds to port 80. The result is that the docker apache process cannot bind to port 80 and fails. (I think.)

The solution may be a command `systemctl stop apache` at the end of the `setup.sh`.

Lastly, I will try to test this branch with a Centos VM tomorrow and post if there are any problems.

TheAspens commented:
Which part of the setup.sh triggered the install of apache?  Apache isn't needed so I'm not sure why it is getting installed.

drshawnkwang commented:
> Which part of the setup.sh triggered the install of apache? 

It turns out the answer is: the PHP package that installs apacge, i.e., what I am 'asking for'. `apt install php` installs apache2 on Debian 9 stretch. 

I was also able to test out this branch on a vanilla Centos 7 VM. First, php does **not** install apache2. 

However, Centos 7 does not build the docker stack. I get the following error.

```
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["docker-compose", "build", "--no-cache"], "delta": "0:00:00.495324", "end": "2019-04-24 14:30:19.727193", "msg": "non-zero return code", "rc": 1, "start": "2019-04-24 14:30:19.231869", "stderr": "Building mysql\nService 'mysql' failed to build: Error parsing reference: \"mariadb:10.3.8-bionic AS mysql\" is not a valid repository/tag: invalid reference format", "stderr_lines": ["Building mysql", "Service 'mysql' failed to build: Error parsing reference: \"mariadb:10.3.8-bionic AS mysql\" is not a valid repository/tag: invalid reference format"], "stdout": "Step 1/3 : FROM mariadb:10.3.8-bionic AS mysql", "stdout_lines": ["Step 1/3 : FROM mariadb:10.3.8-bionic AS mysql"]}
```

reformatting the stederr lines to be more readable:

  "stderr_lines": [
    "Building mysql",
    "Service 'mysql' failed to build: Error parsing reference: \"mariadb:10.3.8-bionic AS mysql\" is not a valid repository\/tag: invalid reference format"
  ],
